athena-dynamodb/src/main/java/com/amazonaws/athena/connectors/dynamodb/util/DDBTypeUtils.java

```diff
@@ -90,14 +92,8 @@ else if (value instanceof BigDecimal) {
         else if (value instanceof List || value instanceof Set) {
             Field child = null;
             if (((Collection) value).isEmpty()) {
-                try {
-                    Object subVal = ((Collection) value).getClass()
-                            .getTypeParameters()[0].getGenericDeclaration().newInstance();
-                    child = inferArrowField(key + ".child", subVal);
-                }
-                catch (IllegalAccessException | InstantiationException ex) {
-                    throw new RuntimeException(ex);
-                }
+                logger.warn("Automatic schema inference encountered empty List or Set {}. Unable to determine element types. Falling back to VARCHAR representation", key);
+                child = inferArrowField("", "");
```

Comment:
we have almost this exact same bug in DocDB. We should think about how to centralize some of this logic. It might be challenging since all the types are different but an abstract with key method to implement per-connector (e.g. 'isEmptyList') might work. Its out of scope for your fix but wanted to call it out.

Reply:
good call out, created https://github.com/awslabs/aws-athena-query-federation/issues/218